scripts/new-chain.ts

```diff
@@ -0,0 +1,187 @@
+import fs from 'fs'
+import path from 'path'
+import { ethers } from 'ethers'
+import dotenv from "dotenv";
+import { ScriptError, ScriptErrorCode, runScript } from './utils';
+import { ADDRESS, CODEHASH, FACTORY_BYTECODE, SIGNER } from './constants';
+
+dotenv.config()
+
+async function newChainWrapper() {
+	const summary: Record<string, unknown> = {};
+	summary.SUCCESS = false
+	summary.LABEL_OPERATION = "--remove-label"
+	try {
+		await verifyNewChainRequest(summary)
+	} finally {
+		const summaryFile = process.env.SUMMARY_FILE
+		if (summaryFile) {
+			fs.writeFileSync(summaryFile, JSON.stringify(summary, null, 2))
+		} else {
+			console.log(summary)
+		}
+	}
+}
+
+async function verifyNewChainRequest(summary: Record<string, unknown>) {
+	const issueBody = process.env.ISSUE_BODY
+	// Extract the RPC URL (first URL) from the issue body as a string
+	const rpcUrl = issueBody?.match(/https?:\/\/[^\s]+/g)?.[0]
+	if (!rpcUrl) {
+		summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.RPC_URL_NOT_FOUND)
+		throwNewChainError(ScriptErrorCode.RPC_URL_NOT_FOUND, summary.ERROR as string)
+	}
+
+	const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
+	const { chainId } = await provider.getNetwork()
+	summary.chainId = chainId
+	const filePath = path.join(__dirname, "..", "artifacts", `${chainId}`, "deployment.json")
+
+	// Check if the factory is already deployed
+	// If the file exists, it means the factory is already deployed
+	// If the file does not exist, it means the factory is not deployed or not added to the repository
+	// and we can proceed with the deployment
+	const deployed = fs.existsSync(filePath)
+	if (deployed) {
+		summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.FACTORY_ALREADY_DEPLOYED)
+		throwNewChainError(ScriptErrorCode.FACTORY_ALREADY_DEPLOYED, summary.ERROR as string)
```

Comment:
We are using error codes from an `enum` - so we can actually compute the summary in the wrapper function.

Reply:
Nice idea, updated 👍🏾